llvm/include/llvm/CodeGen/GlobalISel/GIMatchTableExecutor.h

```diff
@@ -635,8 +636,12 @@ class GIMatchTableExecutor {
 
   bool shouldOptForSize(const MachineFunction *MF) const {
```

Comment:
I don't this this is from your change, but it's a bit confusing how `shouldOptForSize` differs from llvm::shouldOptimizeForSize. I guess this is specifically used for GISel, and as long as the current processing block is optimized for size, we consider this function optimized for size? But don't we derive this check for a block by checking its parent function? Basically, my understanding is that even if `shouldOptimizeForSize` for a function is false, shouldOptimizeForSize for a block that belongs to the function may be true. Here, it doesn't seem like that's the case.

Reply:
Yes, I would like to remove this too, but this caused some build errors because of generated code. In fact, some implementors don't even check PGO.
https://github.com/llvm/llvm-project/blob/66bbbf2e951a486f972a5a54378e6791193ade60/llvm/include/llvm/CodeGen/FastISel.h#L520-L523

Since I'm less familiar with ISel, I thought I'd leave this alone for now.

Reply:
It appears that ISel generally prioritizes optimization for size. Since the `shouldOptForSize` derives its result from the current processing block, the outcome may not be consistent depending on which blocks are being processed, which is counterintuitive.
Upon examining another usage of DAGISel below, I believe this API is actually asking whether the current function should be optimized for size based on the current processing block or DAG.
https://github.com/llvm/llvm-project/blob/66bbbf2e951a486f972a5a54378e6791193ade60/llvm/include/llvm/CodeGen/SelectionDAGISel.h#L396-L398

So, we should keep this way on this context.